panaroo/integrate.py

```diff
@@ -0,0 +1,272 @@
+    
+    for adj in single_gml._adj:
+        for x in single_gml._adj[adj]:
+            y = single_gml._adj[adj][x]
+        
+            y.pop('members')
+            zero = {'members': 0}
+            y.update(zero)
+        
+            genomes = {'genomeIDs' : '0'}
+            y.update(genomes)
+        
+    for node in single_gml._node:
+        y = single_gml._node[node]
+        y.pop('members')
+        
+        zero = {'members': 0} #members are assigned intbitset[0]. needs to be 0
+        y.update(zero)
+        
+        rep = "[']"
+        for char in rep:
+            y['centroid'] = (str(y['centroid'])).replace(char, '')
+            y['dna'] = (str(y['dna'])).replace(char, '')
+            y['protein'] = (str(y['protein'])).replace(char, '')
+        
+        y['hasEnd'] = int(y['hasEnd'])
+        y['mergedDNA'] = int(y['mergedDNA'])
+        y['paralog'] = int(y['paralog'])
+        
+        y['longCentroidID'] = list(y['longCentroidID'])
+        y['seqIDs'] = list(y['seqIDs'])
+    
+    single_gml.graph.update({'isolateNames' : 'x'}) # isolateName assigned x as list(x) in merged graph
+    nx.write_gml(single_gml, output_dir + "final_graph.gml")
+    
+    return single_gml
+
+
+def main(): #Takes a single GFF input, generates a graph and merges with a pre-existing graph
+    args = get_options()
+
+    # make sure trailing forward slash is present
+    args.output_dir = os.path.join(args.output_dir, "")
+    
+    # Create temporary directory
+    temp_dir = os.path.join(tempfile.mkdtemp(dir=args.output_dir), "")
+     
+    directories = args.input_dir + ' ' + temp_dir
+
+    gff_file = []
+    gff_file.append(args.input_gff)
+            
+    filename = (str(args.input_gff).split('/')[-1]).split('.')[0]
+ 
+    process_prokka_input(gff_list = gff_file, 
+                         output_dir = temp_dir, 
+                         quiet = args.quiet, 
+                         n_cpu = args.n_cpu) 
+    
+    cd_hit_out = temp_dir + "/combined_protein_cdhit_out.txt"
+    
+    run_cdhit(input_file=temp_dir + "/combined_protein_CDS.fasta",
+              output_file=cd_hit_out,
+              id=args.id,
+              quiet=args.quiet,
+              n_cpu=args.n_cpu)
+    
+    single_gml, centroid_contexts_single, seqid_to_centroid_single = generate_network(cluster_file=cd_hit_out + ".clstr",
+                                                                                    data_file=temp_dir + "/gene_data.csv",
+                                                                                    prot_seq_file=temp_dir + "/combined_protein_CDS.fasta",
+                                                                                    all_dna=args.all_seq_in_graph)
+    
+    reformat_network(single_gml = single_gml,
+                     output_dir = temp_dir,
+                     isolateName = filename)
+    
+    merge_command = 'panaroo-merge -d ' + directories
```

Comment:
It would probably be better, as this is part of the same package, to instead import the needed functions for the merge rather than going through subprocess (as you have done with run_cdhit etc), as it's less liable to failure, you can control the return types better, and you will get proper error messages if the format of the function called changes

Reply:
Wrote like this initially as panaroo-merge calls on a main() function. Rewrote this as a function within integrate so subprocess no longer needed.